backend/schema/validate.go

```diff
@@ -326,6 +331,65 @@ func ValidateModule(module *Module) error {
 	return errors.Join(merr...)
 }
 
+func findDuplicateDecls(decls []Decl) []error {
+	merr := []error{}
+
+	configs := make(map[string][]int)
+	secrets := make(map[string][]int)
+	dbs := make(map[string]int)
+
+	for i, d := range decls {
+		c, ok := d.(*Config)
+		if ok { // found a config
+			if _, ok := configs[c.Name]; !ok { // no matches possible
+				configs[c.Name] = []int{i}
+				continue
+			}
+			matchIdx, foundMatch := ftlslices.Find(configs[c.Name], func(j int) bool {
```

Comment:
What's the purpose of this? Can't the maps just be `map[string]*Config`, and if one already exists you report it as an error?